server/src/main/java/io/druid/server/coordinator/BalancerStrategy.java

```diff
@@ -31,5 +33,10 @@
 
   BalancerSegmentHolder pickSegmentToMove(List<ServerHolder> serverHolders);
 
+  default Iterator<ServerHolder> pickServersToDrop(DataSegment toDropSegment, NavigableSet<ServerHolder> serverHolders)
```

Comment:
This should have some javadoc explaining what the returned iterator should be. It's non-obvious: the "obvious" return value would be a set of servers to drop from, where the order doesn't matter. But the actual return value is ordered by preferredness, and may contain more servers than we actually want to drop from.

Reply:
I went ahead and added javadoc for whole interface, lmk if I got anything wrong